locking.go

```diff
@@ -243,3 +264,18 @@ func NewLockInstance(ctxParent context.Context, accountName, accountKey, lockNam
 
 	return lockInstance, nil
 }
+
+// IsValidLockName checks if the lock name is between 3-58 (63 minus 5char prefix used) characters long
+// and matches this regex @"^[a-z0-9]+(-[a-z0-9]+)*$"
+func IsValidLockName(lockName string) (bool, error) {
+	lockName = strings.ToLower(lockName)
+	if len(lockName) < 3 || len(lockName) > 58 {
+		return false, fmt.Errorf("lock name: %s must be between 3 and 58 characters long", lockName)
+	}
+
+	if match, _ := regexp.MatchString("^[a-z0-9]+(-[a-z0-9]+)*$", lockName); !match {
```

Comment:
If this is a static regex and you're going to ignore the error anyway - it'd probably be cleaner to compile the regex so we get compile-time parser errors rather than handling runtime errors.

Reply:
Something like this?

```golang 

// validLockNameRegex is a regex used to check the chars are valid as an Azure Storage container name
var validLockNameRegex = regexp.MustCompile("^[a-z0-9]+(-[a-z0-9]+)*$")

// IsValidLockName checks if the lock name is between 3-58 (63 minus 5char prefix used) characters long
// and matches this regex @"^[a-z0-9]+(-[a-z0-9]+)*$"
func IsValidLockName(lockName string) (bool, error) {
	lockName = strings.ToLower(lockName)
	if len(lockName) < 3 || len(lockName) > 58 {
		return false, fmt.Errorf("lock name: %s must be between 3 and 58 characters long", lockName)
	}

	if !validLockNameRegex.MatchString(lockName) {
		return false, fmt.Errorf("lock name: %s must be alphanumberic with no characters other than '-' (regex '^[a-z0-9]+(-[a-z0-9]+)*$')", lockName)
	}

	return true, nil
}
```